src/lib/PlacesProvider.js

```diff
@@ -1,36 +1,40 @@
 /* eslint-disable react/no-unused-state */
 import React, { Component } from 'react'
-import { guid } from 'utils'
 import P from 'prop-types'
+import { read, create } from 'lib/xhr-helpers'
 
 const normalizeItem = ({ name, description, posLat, posLong }) => ({
   name,
   description,
   position: { posLat, posLong },
 })
+const readPlaces = () => read('http://localhost/api/places')
+
+const createPlace = item => create('http://localhost/api/places/create', item)
 
 export const PlacesContext = React.createContext()
 
 export class PlacesProvider extends Component {
   constructor(props) {
     super(props)
-
-    const cachedPlaces = localStorage.getItem('places')
     this.state = {
-      places: JSON.parse(cachedPlaces) || [],
+      places: [],
       onSubmit: this.onPlaceFormSubmit,
     }
   }
+  async componentDidMount() {
+    const places = await readPlaces()
+    this.setState({ places })
+  }
 
-  onPlaceFormSubmit = item => {
-    const newItem = normalizeItem(item)
+  onPlaceFormSubmit = async item => {
+    let newItem = normalizeItem(item)
```

Comment:
Don't reuse variables. Prefer to create a new one when you do a modification on them. You can call `await createPlace(normalizeItem(item))` if you want to spare a variable